operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -168,13 +182,61 @@ private String getLockName(String namespace, String name) {
 
         Future<Void> handler = withLock(reconciliation, LOCK_TIMEOUT_MS, () -> {
             T cr = resourceOperator.get(namespace, name);
+
             if (cr != null) {
-                validate(cr);
+                List<Condition> unknownAndDeprecatedConditions = validate(cr);
+                Promise<Void> createOrUpdate = Promise.promise();
+
+                if (cr.getSpec() == null)   {
+                    InvalidResourceException exception = new InvalidResourceException("Spec cannot be null");
+
+                    S status = createStatus();
+                    Condition errorCondition = new ConditionBuilder()
+                            .withLastTransitionTime(StatusUtils.iso8601Now())
+                            .withType("NotReady")
+                            .withStatus("True")
+                            .withReason(exception.getClass().getSimpleName())
+                            .withMessage(exception.getMessage())
+                            .build();
+                    status.addCondition(errorCondition);
+
+                    log.error("{}: {} spec cannot be null", reconciliation, cr.getMetadata().getName());
+                    updateStatus(reconciliation, status, unknownAndDeprecatedConditions).onComplete(statusResult -> {
```

Comment:
I renamed it from `statusResult` to `notUsed` to make it clear it is intentional. Not sure I can do any better there since regardless the result we always raise the original exception.